webextension/js/popup.js

```diff
@@ -141,6 +141,58 @@ const Logic = {
     return this._currentIdentity;
   },
 
+  cookieStoreId(userContextId) {
+    return `firefox-container-${userContextId}`;
+  },
+
+  _containerTabIterator(userContextId, cb) {
+    browser.tabs.query({
+      cookieStoreId: Logic.cookieStoreId(userContextId)
+    }).then((tabs) => {
+      tabs.forEach((tab) => {
+        cb(tab);
+      });
+    });
+  },
+
+  _containers(userContextId) {
+    return browser.tabs.query({
+      cookieStoreId: Logic.cookieStoreId(userContextId)
+    });
+  },
+
+  removeIdentity(userContextId) {
+    const eventName = "delete-container";
+    if (!userContextId) {
+      return Promise.reject("removeIdentity must be called with userContextId argument.");
+    }
+
+    browser.runtime.sendMessage({
+      method: "sendTelemetryPayload",
+      event: eventName,
+      userContextId
+    });
+
+    const removeTabsPromise = Logic._containers(userContextId).then((tabs) => {
+      const tabIds = tabs.map((tab) => tab.id);
+      return browser.tabs.remove(tabIds);
+    });
```

Comment:
can you just do:

return Logic._containers(userContextId).then((tabs) => { .. })
           .then(() => { ... the rest? })